server/src/queries/mod.rs

```diff
@@ -22,22 +22,50 @@ pub async fn get_latest_blockheight(
 /// By default we get all signals every query.
 /// This is wildly inefficient & we should work towards an alternative.
 /// Since we're essentially only interested in memo's - maybe a way to decode base58 on the DB side?
-pub async fn get_signals(db: &Pool<Postgres>) -> Result<Vec<DBResponse>, sqlx::Error> {
-    sqlx::query_as!(DBResponse,
-        r#"
-        SELECT DISTINCT pk.value as account, uc.memo as memo, uc.nonce as nonce, b.height as height, b.chain_status as "status: BlockStatus", b.timestamp as timestamp
-        FROM user_commands AS uc
-        JOIN blocks_user_commands AS buc
-        ON uc.id = buc.user_command_id
-        JOIN blocks AS b
-        ON buc.block_id = b.id
-        JOIN public_keys AS pk
-        ON uc.source_id = pk.id
-        WHERE uc.type = 'payment'
-        AND uc.source_id = uc.receiver_id
-        AND uc.token = 1
-        AND NOT b.chain_status = 'orphaned'
-        AND buc.status = 'applied'
-        "#
-    ).fetch_all(db).await
+pub async fn get_signals(
+    db: &Pool<Postgres>,
+    timestamp: Option<i64>,
+) -> Result<Vec<DBResponse>, sqlx::Error> {
+    match timestamp {
+        Some(timestamp) => {
+            sqlx::query_as!(DBResponse,
+            r#"
+            SELECT DISTINCT pk.value as account, uc.memo as memo, uc.nonce as nonce, b.height as height, b.chain_status as "status: BlockStatus", b.timestamp as timestamp
+            FROM user_commands AS uc
+            JOIN blocks_user_commands AS buc
+            ON uc.id = buc.user_command_id
+            JOIN blocks AS b
+            ON buc.block_id = b.id
+            JOIN public_keys AS pk
+            ON uc.source_id = pk.id
+            WHERE uc.type = 'payment'
+            AND uc.source_id = uc.receiver_id
+            AND uc.token = 1
+            AND NOT b.chain_status = 'orphaned'
+            AND buc.status = 'applied'
+            AND b.timestamp > $1
+            "#, timestamp
+            ).fetch_all(db).await
+        },
+        None => {
+            sqlx::query_as!(DBResponse,
+            r#"
+            SELECT DISTINCT pk.value as account, uc.memo as memo, uc.nonce as nonce, b.height as height, b.chain_status as "status: BlockStatus", b.timestamp as timestamp
+            FROM user_commands AS uc
+            JOIN blocks_user_commands AS buc
+            ON uc.id = buc.user_command_id
+            JOIN blocks AS b
+            ON buc.block_id = b.id
+            JOIN public_keys AS pk
+            ON uc.source_id = pk.id
+            WHERE uc.type = 'payment'
+            AND uc.source_id = uc.receiver_id
+            AND uc.token = 1
+            AND NOT b.chain_status = 'orphaned'
+            AND buc.status = 'applied'
+            AND b.timestamp > (select extract(epoch from now())) - $1
+            "#, (2629800 * 3) as _
+            ).fetch_all(db).await
+        }
+    }
```

Comment:
Can be shortened to:
```
    sqlx::query_as!(DBResponse,
        r#"
        SELECT DISTINCT pk.value as account, uc.memo as memo, uc.nonce as nonce, b.height as height, b.chain_status as "status: BlockStatus", b.timestamp as timestamp
        FROM user_commands AS uc
        JOIN blocks_user_commands AS buc
        ON uc.id = buc.user_command_id
        JOIN blocks AS b
        ON buc.block_id = b.id
        JOIN public_keys AS pk
        ON uc.source_id = pk.id
        WHERE uc.type = 'payment'
        AND uc.source_id = uc.receiver_id
        AND uc.token = 1
        AND NOT b.chain_status = 'orphaned'
        AND buc.status = 'applied'
        AND b.timestamp > $1
        "#, match timestamp {
            Some(ts) => ts,
            None => 2629800 * 3,
        }
        ).fetch_all(db).await

Reply:
no it cannot. if None is passed in the above code sample, the query will get all blocks until 3 months after the genesis block. we want 3 months before when the query is run, and I figured we would want to calculate "now" on the DB side, hence 2 different queries.

Reply:
No not DB side - calculate it on the server & make that the default. (Still can be shortened then)

Reply:
> Also - this implementation is wrong - `http://localhost:8080/api/v1/test?network=Mainnet` returns `Failed to deserialize query string: invalid digit found in string`.
> 
> I recommend reading the Axum docs for query params handling.

ON the monday.com task, you asked for the timestamp to be passed as query parameter, then later you said in slack this query parameter should be a unix timestamp. The URL in this example does not have a `startDate` query parameter passed in

Reply:
i.E. something like:
```
        "#, match timestamp {
            Some(ts) => ts,
            None => std::time::SystemTime::now()
            .duration_since(std::time::UNIX_EPOCH)
            .unwrap()
            .as_secs() as i64
            - 7889400,
}

Reply:
hey mine's a bit different but it does the same thing!